### PR TITLE
Created a new ShipMenu-widget based on the old ShipTabMenuWidget.

### PR DESCRIPTION
**Pull request by Royber121**
Added a global EventBus.
Made the HUD appear, when a ship is selected.
-Ship is added to group "selected_units"
-Event "selected" is fired via EventBus
-PlayerHUD changes the visibility of the ShipMenu
-ShipMenu creates some inventory slots based on the ships num_of_slots

### CODE REVIEW

**Comment by Royber121** [comment]
Dammit, I hate git! I commited to the wrong branch. Ignore the last commit, when you merge the PR.

File: Assets/UI/Scenes/TabWidget/ShipMenu.gd [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	for idx in range(num_slots):
+		var inventory_slot = inventory_slot_scene.instance()
+		inventory_slot.set_resource_type(idx)
+		grid_container.add_child(inventory_slot)
+	
+func _set_resource_in_slot():
+	pass
+	
+func _ready():
+	pass # Replace with function body.
```
Comment (aaronfranke) [inline Assets/UI/Scenes/TabWidget/ShipMenu.gd:38]: Remove unused method.

File: Assets/UI/Scripts/TabWidget.gd [hunk 2]
```diff
@@ -17,3 +18,24 @@ func _process(_delta: float) -> void:
...
+		# only one Unit, so probably only possible to build etc. if only one Unit selected
+		if selected_unit.get_groups().find("ships") == 1: # Its a ship
+			_set_detail_body(ship_menu)
+			ship_menu.set_selected_ship(selected_unit)
+			_set_detail_visibility(true)			
+		if selected_units[0].get_groups().find("ships") != 1: # Its not a ship. Do something else
+			pass
+	elif selected_units.size() > 1: # multiple units selected, do something else
+		pass
+	pass
```
Comment (aaronfranke) [inline Assets/UI/Scripts/TabWidget.gd:34]: Remove unused `pass`, and consider removing (or commenting out) blocks which only contain `pass`.

File: Assets/UI/Scenes/TabWidget/ShipMenu.gd [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	_add_resource_slots(selected_ship.num_of_slots)
+	
+func _on_ColoniseButton_pressed():
+	pass
+	
+func _set_caption(new_caption):
+	name_caption.text = new_caption
+	
+func _add_resource_slots(num_slots):
+	# first remove all slots from previous selection (if any)
```
Comment (aaronfranke) [inline Assets/UI/Scenes/TabWidget/ShipMenu.gd:23]: Make comments start with a capital letter and end with a period.

**Review by aaronfranke — CHANGES_REQUESTED** [review]
It doesn't seem to work properly:

Screenshot from 2020-03-16 15-29-33

When it comes to Git, read through the [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html) article in the Godot documentation, or if you prefer a graphical Git client, watch [my video on Git workflow using GitKraken](https://www.youtube.com/watch?v=0XKy8YpP5Ww).

**Comment by LinuxDonald** [comment]
@Royber121 any update because of the requested changes?

**Comment by aaronfranke** [comment]
Closing, PR author is inactive.

**Pull request closed without merging**